pkg/ingress/kube/annotations/match.go

```diff
@@ -0,0 +1,299 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package annotations
+
+import (
+	"fmt"
+	"strings"
+
+	. "github.com/alibaba/higress/pkg/ingress/log"
+	networking "istio.io/api/networking/v1alpha3"
+)
+
+const (
+	exact       = "exact"
+	regex       = "regex"
+	prefix      = "prefix"
+	matchMethod = "match-method"
+	matchQuery  = "match-query"
+	matchHeader = "match-header"
+	sep         = " "
+)
+
+var (
+	methodList = []string{"GET", "HEAD", "POST", "PUT", "DELETE", "CONNECT", "OPTIONS", "TRACE", "PATCH"}
+	methodMap  map[string]struct{}
+)
+
+type match struct{}
+
+type MatchConfig struct {
+	Methods     []string
+	Headers     map[string]map[string]string
+	QueryParams map[string]map[string]string
+}
+
+func (m match) Parse(annotations Annotations, config *Ingress, _ *GlobalContext) (err error) {
+	config.Match = &MatchConfig{}
+
+	if err = m.matchByMethod(annotations, config); err != nil {
+		IngressLog.Errorf("parse methods error %v within ingress %s/%s", err, config.Namespace, config.Name)
+	}
+
+	if err = m.matchByHeader(annotations, config); err != nil {
+		IngressLog.Errorf("parse headers error %v within ingress %s/%s", err, config.Namespace, config.Name)
+	}
+
+	if err = m.matchByUrlParam(annotations, config); err != nil {
+		IngressLog.Errorf("parse query params error %v within ingress %s/%s", err, config.Namespace, config.Name)
+	}
+
+	return
+}
+
+func (m match) ApplyRoute(route *networking.HTTPRoute, ingressCfg *Ingress) {
+	// apply route for method
+	config := ingressCfg.Match
+	if config.Methods != nil {
+		for i := 0; i < len(route.Match); i++ {
+			route.Match[i].Method = createMethodMatch(config.Methods...)
+			IngressLog.Debug(fmt.Sprintf("match :%v, methods %v", route.Match[i].Name, route.Match[i].Method))
+		}
+	}
+
+	// apply route for headers
+	if config.Headers != nil {
+		for i := 0; i < len(route.Match); i++ {
+			if route.Match[i].Headers == nil {
+				route.Match[i].Headers = map[string]*networking.StringMatch{}
+			}
+			addHeadersMatch(route.Match[i].Headers, config)
+			IngressLog.Debug(fmt.Sprintf("match headers: %v, headers: %v", route.Match[i].Name, route.Match[i].Headers))
+		}
+	}
+
+	if config.QueryParams != nil {
+		for i := 0; i < len(route.Match); i++ {
+			if route.Match[i].QueryParams == nil {
+				route.Match[i].QueryParams = map[string]*networking.StringMatch{}
+			}
+			addQueryParamsMatch(route.Match[i].QueryParams, config)
+			IngressLog.Debug(fmt.Sprintf("match : %v, queryParams: %v", route.Match[i].Name, route.Match[i].QueryParams))
+		}
+	}
+}
+
+func (m match) matchByMethod(annotations Annotations, ingress *Ingress) error {
+	if !annotations.HasASAP(matchMethod) {
```

Comment:
```suggestion
	if !annotations.HasHigress(matchMethod) {
```
Only the `higress.io` prefix is required.